pkg/cmd/pr/shared/commentable.go

```diff
@@ -84,37 +85,88 @@ func CommentableRun(opts *CommentableOptions) error {
 	if err != nil {
 		return err
 	}
+	opts.Host = repo.RepoHost()
+	if opts.EditLast {
+		return updateComment(commentable, opts)
+	}
+	return createComment(commentable, opts)
+}
 
-	var lastComment *api.Comment
-	if opts.Upsert {
-		username, err := opts.RetrieveCurrentUser()
+func createComment(commentable Commentable, opts *CommentableOptions) error {
+	switch opts.InputType {
+	case InputTypeWeb:
+		openURL := commentable.Link() + "#issuecomment-new"
+		if opts.IO.IsStdoutTTY() && !opts.Quiet {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", text.DisplayURL(openURL))
+		}
+		return opts.OpenInBrowser(openURL)
+	case InputTypeEditor:
+		var body string
+		var err error
+		if opts.Interactive {
+			body, err = opts.InteractiveEditSurvey("")
+		} else {
+			body, err = opts.EditSurvey("")
+		}
 		if err != nil {
 			return err
 		}
-		comments := commentable.CommentsForUser(username)
-		if len(comments) > 0 {
-			lastComment = &comments[len(comments)-1]
+		opts.Body = body
+	}
+
+	if opts.Interactive {
+		cont, err := opts.ConfirmSubmitSurvey()
+		if err != nil {
+			return err
+		}
+		if !cont {
+			return errors.New("Discarding...")
 		}
 	}
 
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	apiClient := api.NewClientFromHTTP(httpClient)
+	params := api.CommentCreateInput{Body: opts.Body, SubjectId: commentable.Identifier()}
+	url, err := api.CommentCreate(apiClient, opts.Host, params)
+	if err != nil {
+		return err
+	}
+
+	if !opts.Quiet {
+		fmt.Fprintln(opts.IO.Out, url)
+	}
+
+	return nil
+}
+
+func updateComment(commentable Commentable, opts *CommentableOptions) error {
+	username, err := opts.RetrieveCurrentUser()
+	if err != nil {
+		return err
+	}
+
+	comments := commentable.CommentsForUser(username)
+	if len(comments) == 0 {
+		return fmt.Errorf("no comments created by %s found", username)
```

Comment:
The error when there is no comment makes it hard to use it from the bot. I and the original issue would like an option to use it from the bot.
If `--edit-last` is not a good name, it would be more convenient to use an option name such as `--upsert` to avoid the error.